server/server.js

```diff
@@ -0,0 +1,76 @@
+import express from "express";
+import * as path from "path";
+import { isCorrectAnswer, randomQuestion, Questions } from "./quiz.js";
+import bodyParser from "body-parser";
+import cookieParser from "cookie-parser";
+import dotenv from "dotenv";
+dotenv.config();
+
+const app = express();
+app.use(bodyParser.json());
+app.use(cookieParser(process.env.COOKIE_SECRET));
+
+app.get("/api/question", (req, res) => {
+  const { id, category, question, answers } = randomQuestion();
+  res.json({ id, question, answers, category });
+});
+
+// callback: (req, res) => {}
+app.post("/api/question", (req, res) => {
+  const { id, answer } = req.body;
+  const score = req.signedCookies.score
+    ? JSON.parse(req.signedCookies.score)
+    : {
+        answered: 0,
+        correct: 0,
+      };
+  const question = Questions.find((q) => q.id === id);
+
+  if (!question) {
+    return res.sendStatus(404);
+  }
+
+  score.answered += 1;
+  if (isCorrectAnswer(question, answer)) {
+    score.correct += 1;
+    res.cookie("score", JSON.stringify(score), { signed: true });
+    return res.json({ result: "correct" });
+  } else {
+    res.cookie("score", JSON.stringify(score), { signed: true });
+    return res.json({ result: "incorrect" });
+  }
+});
+
+app.get("/api/score", (req, res) => {
+  const score = req.signedCookies.score
+    ? JSON.parse(req.signedCookies.score)
+    : {
+        answered: 0,
+        correct: 0,
+      };
+  res.json(score);
+});
```

Comment:
Det er lurt å trekke API-metoder ut i en egen fil å bruke express Router for å gruppere disse